lib/voicemail-admin.js

```diff
@@ -0,0 +1,779 @@
+      }
+
+      var domain = entryComponents[2];
+
+      return dal.context.get(domain)
+        .then(function(context) {
+          if (!context) {
+            throw new Error(sprintf('Context \'%s\' not found.', domain));
+          }
+
+          return dal.mailbox.findByContext(context);
+        })
+        .then(function(mailboxes) {
+          logger.info(sprintf('%-20s %-25s %-25s',
+                              'number',
+                              'email',
+                              'name').underline);
+
+          mailboxes.forEach(function (mailbox) {
+            logger.info(sprintf('%-20s %-25s %-25s',
+              mailbox.mailboxNumber + '@' + domain,
+              mailbox.email,
+              mailbox.name));
+          });
+        });
+    },
+
+    createContext: function(chosenCommand, entryComponents) {
+      if (entryComponents.length !== 3) {
+        return invalidSyntax(entryComponents);
+      }
+
+      var newContextDomain = entryComponents[2];
+      var newContext = dal.context.create(newContextDomain);
+
+      return dal.context.get(newContextDomain)
+        .then(function(existingContext) {
+          if (existingContext) {
+            throw new Error(sprintf('Context with domain \'%s\' already ' +
+                                    'exists.', newContextDomain));
+          }
+
+          return dal.context.save(newContext);
+        })
+        .then(function() {
+          logger.info('Successfully created context \'%s\''.green,
+                      newContextDomain);
+        });
+    },
+
+    createMailbox: function(chosenCommand, entryComponents) {
+      if (entryComponents.length !== 6) {
+        return invalidSyntax(entryComponents);
+      }
+
+      var mailboxKey = entryComponents[2].split('@');
+
+      if (mailboxKey.length !== 2) {
+        return invalidSyntax(entryComponents);
+      }
+
+      var mailboxNumber = mailboxKey[0];
+      var mailboxContext = mailboxKey[1];
+      var password = entryComponents[3];
+      var name = entryComponents[4];
+      var email = entryComponents[5];
+      var holdContext;
+
+      return dal.context.get(mailboxContext)
+        .then(function(context) {
+          holdContext = context;
+
+          if (!context) {
+            throw new Error(sprintf('Context \'%s\' requested for \'%s@%s\' ' +
+                                    'does not exist.', mailboxContext,
+                                    mailboxNumber, mailboxContext));
+          }
+
+          return dal.mailbox.get(mailboxNumber, context);
+        })
+        .then(function(existingMailbox) {
+          if (existingMailbox) {
+            throw new Error(sprintf('Requested mailbox \'%s@%s\' already ' +
+                                    'exists.', mailboxNumber,
+                                    mailboxContext));
+          }
+
+          var mailbox = dal.mailbox.create(mailboxNumber, holdContext);
+          mailbox.password = password;
```

Comment:
You can pass these parameters as part of the call to create. Just pass an object as the third param with keys matching the param names.